### ISSUE TITLE
Accessibility Issues Report

### ISSUE THREAD

**Issue by bnmnetp** [RunestoneInteractive/RunestoneComponents#394]
Hi Brad,
 
Thanks to the information you provided, we were able to do some testing of Runestone with our lab, and as promised, I’m sending over the results. I’ve attached the detailed results here, but here’s a summary:
 
Text & Navigation
1) Actual text reads fine; code & symbols read fine
2) Navigation is problematic—screen reader is not notified when “Mark as Completed” changes and new chapter link appears—also, new link exists navigationally before the button, so user would need to know to backtrack to previous navigational items.
3) Highlight text does not work at all for screen reader (which we guessed)
4) Fields are not all labeled in a way that is logical for a screen reader (details in attached results)
 
Interactive Features
5) In-page quiz questions functional, but user is not notified of right/wrong answer on “Check Me” button
6) ActiveCode & CodeLens operate but screen reader cannot read resulting text (after navigating through many blanks); color coded arrows are not labeled meaningfully—technically functional, but essentially not accessible
 
Image & Video
7) Images have no meaningful alternate text for those who can’t see them (e.g. flow charts)
8) Videos play but button labels are misleading—also, videos are not captioned
 
Our next step is to create some alternative plans if a student with an accommodation request enrolls in the course for fall. We also want to know if your developers might be able to address any of these issues in the coming months (we will create alternate plans regardless, because of how long these kinds of fixes can take).
 
Things like field labels & navigation seem critical and also, at my guess, may be the low-hanging fruit in this list. Alt text might be the same. Looking at this list, do you think your developers might incorporate any of these issues in their future plans?
 
Thanks
 
Rebecca
[User Steps Runestone with A11y Desc.docx](https://github.com/RunestoneInteractive/RunestoneComponents/files/1092029/User.Steps.Runestone.with.A11y.Desc.docx)

**Comment by aerenchyma** [comment]
Thanks!

> Our next step is to create some alternative plans if a student with an accommodation request enrolls in the course for fall. We also want to know if your developers might be able to address any of these issues in the coming months (we will create alternate plans regardless, because of how long these kinds of fixes can take).

Do you know if this proceeded?

I have a student enrolling in the Fall in one of our UMich courses (Paul's server) who has various accessibility concerns for their participation with readings. My understanding is that they don't necessarily always use a screenreader, but they do need conversion to mp3s and other formats through our university's Services for Students with Disabilities office. 

I'm wondering if the PDF conversion (s.t. there could be accessible PDFs) is a stopgap measure, or if Rebecca et al have other thoughts.

@presnick ,@soney, I'm interested in your thoughts here, too. Have you had any students enrolled who needed accessibility features like these? How did you help with that?
